add to path instructions (#714)
Agoric/documentation: main/getting-started/before-using-agoric.md

```shell
node --version # 14.15.0 or higher
npm install --global yarn

> git clone -b community-dev https://github.com/Agoric/agoric-sdk # "community-dev" branch

Reviewer: The scope of this PR is drifting. Please be sure to keep the title and description up to date.

Alternatively, make a separate PR regarding `-b community-dev`.

cd agoric-sdk
yarn install
yarn build
yarn link-cli ~/bin/agoric # creates an Agoric directory to reference
agoric --version # should print the version number of the SDK
```

Now you are ready proceed to [starting a project](/getting-started/start-a-project.md).

> If `agoric` is not found, then you may need to add the parent directory of where your `yarn link-cli` command created a folder. For example, if you created a folder called `Agoric` in your `/bin` directory, you may need to add `/bin` to your PATH.

Reviewer: also (less citical):

Should we tell them _how_? as in `export PATH=~/bin:$PATH`? and/or link to some widely-used docs on the topic?

 - [Linux add to $PATH: Fix "command not found" error \(Linux & Mac\) \- YouTube](https://www.youtube.com/watch?v=gkqsLRDnqlA)

This presentation is a good overview of the Agoric SDK setup process,
though a few details are out of date:

- node version: 12.x is too old; use 14.15.0 or higher
- skip `git checkout hackathon-2020-11`; use the default `master` branch

<iframe width="560" height="315" src="https://www.youtube.com/embed/w0By22jYhJA" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>

**Note:** Video omits adding the Agoric SDK to your PATH.
:::

## Platform: Linux shell or equivalent

The Agoric SDK is supported on
<a href="https://en.wikipedia.org/wiki/Linux">Linux</a>,
<a href="https://www.apple.com/macos/">MacOS</a>,
<a href="https://docs.microsoft.com/en-us/windows/wsl/">Windows Subsystem for Linux (WSL)</a>, and
<a href="https://www.docker.com/">Docker</a>.

- To open a terminal on MacOS, see **Applications>Utilities>terminal** in the **Finder**.
- To launch a terminal on Linux, use the **Terminal** application.
- To access WSL from Windows, visit the [WSL documentation](https://docs.microsoft.com/en-us/windows/wsl/).

> - To use a Docker container, see the [Docker documentation](https://docs.docker.com/) and the [Agoric container images](https://hub.docker.com/u/agoric).

Reviewer: Based on discussion with @michaelfig , I'm not sure we want to support customers in using those docker images for development. We sort of supported them during our beta release for clients / wallets for folks that didn't want to install the SDK. But these docs are not for that audience.

## Install Node.js 14.15.0 or higher

> Download from [NodeJS](https://nodejs.org/) and follow the instructions for your platform.

Reviewer: They call themselves "Node.js"; any reason not to use that name?

cd agoric-sdk
yarn install
yarn build
```

> ### Add the `agoric` command to your PATH

Reviewer: Adding a whole section here is somewhat awkward.

How about we just add:

```
export PATH=~/bin:$PATH
```

before `agoric --version` in this section, and then move the whole dedicated section on PATH to the expanded directions below?

Author: sounds good! thanks!

The `agoric` command is a shell script that uses `yarn` to run the Agoric SDK from the `agoric-sdk` directory. To use it from any directory on MacOS, add it to your path:

```shell

> yarn link-cli ~/bin/agoric # creates an Agoric directory to reference

Reviewer: That assumes `~/bin` exists. And this step doesn't create any directories. How about:

```suggestion
mkdir -p ~/bin
yarn link-cli ~/bin/agoric # installs a small "agoric" script
```

Reviewer: How was this resolved? I'm struggling to find it.

echo $PATH # show current PATH

> export PATH=$PATH:/Users/*USERNAME*/bin # adds the parent directory to your PATH

Reviewer: ```suggestion
export PATH=$PATH:$HOME/bin # adds the parent directory to your PATH
```

echo $PATH # show updated PATH to include the parent directory of Agoric folder

> # RESTART YOUR SHELL FOR CHANGES TO TAKE EFFECT #

Reviewer: Restarting your shell will make these changes disappear, I think.

Author: you're right one command adds it to the shell and then it'll disappear on restart... adding it permanently to your shell by modifying the ".bash_profile" file requires a restart.

Reviewer: right... discussion of adding it permanently should go in the expanded section on PATH handling... which... btw... where did it go?

Author: sorry i removed the expanded section on PATH handling based on the comment that it was [somewhat awkward](https://github.com/Agoric/documentation/pull/714/files/c2282953f685c7cba759ae42bb221d0057a16a46#r965208761) so I consolidated and married them... you'd like me to revive that section and place it near the bottom with the expanded "Install `agoric` CLI" instructions?

Reviewer: That was what i was suggesting, yes; I'm not sure it's a good suggestion, though. I'm juggling too much to give this the attention it deserves just now.

Any thoughts, @kennyrowe?

Author: Any opinions? I've got a video demonstrating [install here](https://youtu.be/Yp76WxbaNhQ).  [@kennyrowe ](https://github.com/kennyrowe) @turadg @dckc

Reviewer: This is to replace the existing video explainer? I don't see a change to that effect in the PR.
The existing video has voice-over; I'd rather not lose that.

Author: i haven't slotted in the video... it's not deployed on Agoric channels yet that sample is on my personal YouTube set to private unless you have the link.

Author: @dckc I've amended the instructions - think we could resolve this conversation, push these changes across, and then I can create an up to date video with voiceover reflecting these awesome changes we've made? =)

agoric --version # should print the version number of the SDK
```

Now you are ready proceed to [starting a project](/getting-started/start-a-project.md).

> If `agoric` is not found, then you may need to add the parent directory of where your `yarn link-cli` command created a folder. For example, if you created a folder called `Agoric` in your `/Users/bin` directory, you may need to add `/Users/bin` to your PATH.

Reviewer: ```suggestion
If `agoric` is not found, then you may need to add the parent directory of where your `yarn link-cli` command created a folder. For example, if you created a folder called `Agoric` in your `~/bin` directory, you may need to add `~/bin` to your PATH.
```

```shell
agoric --version # v0.16.0 "community-dev" branch
```

If the install was successful, you are ready to proceed to [starting a project](/getting-started/start-a-project.md).

Should `agoric` not be found, then you may need to add the parent directory of where your `yarn link-cli` command created a "agoric" directory.

> Please see the [Linux add to \$PATH: Fix "command not found" error (Linux & Mac) - YouTube](https://www.youtube.com/watch?v=gkqsLRDnqlA) for more information.

Reviewer: It's a little odd that we discuss adding to PATH only in the quickstart section. The premise of the quickstart section is that it's for people who know the basics.

Adding a `export PATH=~/bin:$PATH` line in the code box seems more straightforward. Then we could omit these two paragraphs or move them down where we have "Note: Run echo $PATH to see your current $PATH."

not critical I suppose.

Author: thanks for approving the PR! regardless of the debate I think it's in superior condition! =) 

so if look we look at the greats like installing Node.js then it automatically links for us. Also, our VS Code extension is handling it beautifully. Ideally, I'd like to see us get to a place in which the `yarn link-cli` command is optional. 

That said, the instructions with an `export PATH=~/bin:$PATH` creates a new "bin" directory and adds more complexity to the $PATH. It's messy. After doing this several times, I think running `echo $PATH`, finding your local bin, and adding `agoric` in that location is optimal imo.

Reviewer: > After doing this several times, I think running echo $PATH, finding your local bin, and adding agoric in that location is optimal imo.

Then shouldn't the quickstart section do likewise?
